db.go

```diff
@@ -383,11 +383,15 @@ func (db *DB) close() error {
 	if !db.opened {
 		return nil
 	}
-	
+
 	db.opened = false
 
 	db.freelist = nil
-	db.path = ""
+
+	// defer clearing the path because funlock may need it.
+	defer func() {
+		db.path = ""
+	}()
```

Comment:
Can you move this down to just before the `return nil` at the end of the function? I'd rather not have a `defer` if it's not necessary.